src/lib/calls.ts

```diff
@@ -0,0 +1,232 @@
+            throw new Error(`HTTP error! Status: ${response.status}`);
+        }
+
+        const data = await response.json();
+        return data;
+    } catch (error) {
+        console.error('Error fetching scooters:', error);
+        return { success: false };
+    }
+}
+export async function updateScooter(scooterId: string, latitude: number, longitude: number, battery: number, needRepairs: boolean, checkedOut: boolean) {
+    try {
+        const response = await fetch(APIS, {
+            method: 'PATCH',
+            headers: {
+                'Content-Type': 'application/json',
+            },
+            body: JSON.stringify({ scooterId, latitude, longitude, battery, needRepairs, checkedOut })
+        });
+
+        if (!response.ok) {
+            throw new Error(`HTTP error! Status: ${response.status}`);
+        }
+
+        const data = await response.json();
+        return data;
+    } catch (error) {
+        console.error('Error fetching scooters:', error);
+        return { success: false };
+    }
+}
+export async function addUser(firstname: string, lastname: string, email: string, passwordHash: string, isAdmin: boolean | null) {
+    try {
+        const response = await fetch(APIU, {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+            },
+            body: JSON.stringify({ firstname, lastname, email, passwordHash, isAdmin })
+        });
+
+        if (!response.ok) {
+            throw new Error(`HTTP error! Status: ${response.status}`);
+        }
+
+        const data = await response.json();
+        return data;
+    } catch (error) {
+        console.error('Error fetching scooters:', error);
+        return { success: false };
+    }
+}
+export async function deleteUser(userId: string) {
+    try {
+        const response = await fetch(APIU, {
+            method: 'DELETE',
+            headers: {
+                'Content-Type': 'application/json',
+            },
+            body: JSON.stringify({ userId })
+        });
+
+        if (!response.ok) {
+            throw new Error(`HTTP error! Status: ${response.status}`);
+        }
+
+        const data = await response.json();
+        return data;
+    } catch (error) {
+        console.error('Error fetching scooters:', error);
+        return { success: false };
+    }
+}
+export async function getUser() {
+    try {
+        const response = await fetch(APIU, {
+            method: 'GET',
+            headers: {
+                'Content-Type': 'application/json',
+            }
+        });
+
+        if (!response.ok) {
+            throw new Error(`HTTP error! Status: ${response.status}`);
+        }
+
+        const data = await response.json();
+        return data;
+    } catch (error) {
+        console.error('Error fetching scooters:', error);
+        return { success: false };
+    }
+}
+export async function addTransaction(customerId: string, scooterId: string) {
```

Comment:
Should this automatically set checkedOut to true?

Reply:
Do we also store failed transactions.

Reply:
I separated transactions from rentals